### DISCUSSION TITLE
Set up scheduled Debian/Ubuntu unattended upgrades

### DISCUSSION THREAD

**Discussion by giscus** [FarisZR/FarisZR.com#10] [Comments]
# Set up scheduled Debian/Ubuntu unattended upgrades

When you run a server that relies on containers for almost everything, keeping the system up to date has almost zero chance of breaking anything. So here is a quick guide to setting up automatic updates on a fixed schedule + automatic reboots for kernel updates (optional) installing unattended-upgrades 1 2 sudo apt update sudo apt install unattended-upgrades Configuring Unattended-upgrades Depending on what kind of updates you feel comfortable applying, you can customize the source repository as you like.

https://fariszr.com/setup-scheduled-debian-ubuntu-unattended-upgrades/

**Comment by Egwor** [comment]
This was very useful. You could do a few things differently, which might be interesting.

*  You can get a job to run after the other. Use the After like [1]
* It is best to radomise when the jobs run so the RadomizedDelaySec is a good thing to do since it puts less strain on the infrastructure.

[1]Example after

[Unit]
Description=Daily apt upgrade and clean activities
After=apt-daily.timer

[Timer]
..
RandomizedDelaySec=60m
...
